src/vcpkg/base/files.cpp

```diff
@@ -3082,12 +3122,17 @@ namespace vcpkg
                 return false;
             }
 
-            if (options == CopyOptions::overwrite_existing)
+            if (options == CopyOptions::overwrite_existing || options == CopyOptions::update_existing)
             {
                 destination_fd.ftruncate(0, ec);
                 if (ec) return false;
             }
 
+            if (options == CopyOptions::update_existing && destination_stat.st_mtime >= source_stat.st_mtime)
+            {
+                return false;
+            }
+
```

Comment:
My reading of these lines is that:
1. When passing `update_existing`, the file will always be truncated to 0 bytes
2. Then, timestamps will be compared. If the file is new enough, return and leave the file _empty_.

Should these two conditionals be swapped (3131 first, then 3125)? It would be great to get some tests.